---
tags:
  - Status/InProgress
links:
  - "[[Project]]"
---
Transfer Learning from [[Natural Language Processing]], [[Machine Learning Workflow]]

# Problem definition

Pros

Ban Ä‘áº§u lÃ  chuá»—i ADN Ä‘Æ°á»£c cáº¯t nhá» thÃ nh tá»«ng Ä‘oáº¡n read nhá», tá»« nhá»¯ng Ä‘oáº¡n contig sáº½ Ä‘Æ°á»£c xÃ¡c Ä‘á»‹nh overlap Ä‘á»ƒ táº¡o lÃªn contig. Tiáº¿p tá»¥c overlap chÃºng ta sáº½ táº¡o Ä‘Æ°á»£c assembly graph bao gá»“m cÃ¡c liÃªn káº¿t giá»¯a cÃ¡c contig

CÃ³ nhá»¯ng thÃ´ng tin tá»« nhá»¯ng plasmid Ä‘Ã£ tÃ¬m Ä‘Æ°á»£c, nhá»¯ng thÃ´ng tin liÃªn quan Ä‘áº¿n plasmid vÃ  chromosome

Cons

- Contigs cÃ³ nhá»¯ng Ä‘oáº¡n ráº¥t ngáº¯n, khÃ´ng cÃ³ liÃªn káº¿t vá»›i cÃ¡c contig khÃ¡c vÃ  khÃ´ng xÃ¡c Ä‘inh Ä‘Æ°á»£c cÃ¡c giÃ¡ trá»‹ nhÆ° GC, ...
- Äoáº¡n chá»©a thÃ´ng tin cÃ³ thá»ƒ bá»‹ cáº¯t ngáº¯n ra lÃ m khÃ´ng xÃ¡c Ä‘á»‹nh Ä‘Æ°á»£c hoáº·c lÃ  Ä‘oáº¡n cÃ³ lá»—i trÃªn quÃ¡ trÃ¬nh assembly

### Sequence assembly

# System Design

## Input

Input Ä‘áº§u vÃ o lÃ  file cÃ³ thá»ƒ lÃ  fasta
## Function

Processing to assembly - QC - Extract features - Query - Classify with multiple model to increase confidence - Testing with existing model

## Output

Label káº¿t quáº£ tá»«ng contig theo 3 nhÃ³m p c ambigous

(Optional) Rebuild data

# Data 

Start with data from 

# Workflow

# Data collection

## Data storage

(Optional) Store data in graph database or vector database
## QC

QC giá»‘ng nhÆ° tiá»n xá»­ lÃ½ 









































- Thu tháº­p dá»¯ liá»‡u tÃ¬m táº¥t cáº£ complete genomes cá»§a má»™t loÃ i tá»« cÆ¡ sá»Ÿ dá»¯ liá»‡u GeneBank
	- E. coli
	- K. pneumoniea
	- S. aureus
- XÃ¡c Ä‘á»‹nh Ä‘Æ°á»£c sequence nÃ o lÃ  chromosome vÃ  plasmid
- MÃ´ phá»ng láº¡i quÃ¡ trÃ¬nh giáº£i trÃ¬nh tá»± gene sá»­ dá»¥ng ART
- Cháº¡y genome assembly trÃªn dá»¯ liá»‡u mÃ´ phá»ng sáº½ táº¡o ra assembly graph
- Map contig vÃ o trong genome thÃ¬ mÃ¬nh sáº½ biáº¿t Ä‘Æ°á»£c nguá»“n gá»‘c 






á»¨ng dá»¥ng transformer vÃ o trong bÃ i toÃ¡n cá»¥ thá»ƒ lÃ  plasmid

1. Tá»« bá»™ gene hoÃ n chá»‰nh
2. Táº¡o thÃ nh cÃ¡c contig tá»« short-read
3. Chuyá»ƒn Ä‘á»•i giá»¯a assembly graph vÃ  dáº¡ng Ä‘á»™c láº­p fasta
4. Alignment cÃ¡c contig vÃ o database Ä‘á»ƒ xÃ¡c Ä‘á»‹nh Ä‘Ã¢u lÃ  contig vá»›i Ä‘á»™ chÃ­nh xÃ¡c cao (Hiá»‡n táº¡i theo mÃ¬nh biáº¿t thÃ¬ cÃ¡c phÆ°Æ¡ng phÃ¡p align chá»§ yáº¿u táº­p trung vÃ o nháº­n diá»‡n cÃ¡c gene trÃªn plasmid))
    
    <aside> ğŸ’¡ ThÃªm cÃ¡ch Ä‘á»ƒ nháº­n diá»‡n gene trÃªn chromosome
    
    </aside>
    
5. Tá»« thÃ´ng tin liÃªn quan Ä‘á»ƒ xÃ¡c Ä‘á»‹nh thÃªm Ä‘Ã¢u lÃ  contig
6. Tokenize thÃ´ng tin liÃªn quan Ä‘áº¿n plasmid vÃ  liÃªn quan Ä‘áº¿n chromosome (PC)
7. TÃ¬m ra cÃ¡c gene Ä‘áº·c thÃ¹ cho vi khuáº©n báº±ng cÃ¡ch tokenize tá»« Transformer
8. MÃ´ phá»ng láº¡i bá»™ gene hoÃ n chá»‰nh cá»§a vi khuáº©n (Äoáº¡n CHR or PLM Ä‘Ãºng ra thÃ¬ pháº£i cÃ³ 2 máº¡ch, tÃ¬m váº¿ tÆ°Æ¡ng á»©ng Ä‘á»ƒ sá»­a nhá»¯ng Ä‘iá»ƒm sai)
9. Test
    1. Cháº¡y dá»¯ liá»‡u trÃªn platon vÃ  PLASMe

---

### Analysis

306 edges

- chromosome,chromosome 109
- chromosome,plasmid 0
- chromosome,unlabeled: 70
- chromosome,ambiguous 4

- plasmid,chromosome 0
- plasmid,plasmid 51
- plasmid,ambiguous 2
- plasmid,unlabeled 24

- unlabeled,unlabeled 12
- unlabeled,plasmid 6
- unlabeled,chromosome 20
- unlabeled,ambiguous 0

- ambiguous,ambiguous 1
- ambiguous,plasmid
- ambiguous,chromosome 1
- ambiguous,unlabeled 4

# Abstract

Most of the task treat contig as independent, but the relation (edge) connect with different contig (had detected as plasmid is also important information)

- Classification of a contig can be improved from the knowledge of the classification of the neighboring contigs in the assembly graph

1. Check Ä‘á»™ chÃ­nh xÃ¡c cá»§a viá»‡c align, mÃ¬nh cÃ³ thá»ƒ tÄƒng ngÆ°á»¡ng
2. Tá»« SPdes, IUnicycler v0.5.0 (Wick et al., 2017) and SKESA v2.4.0 (Souvorov et al., 2018), two widely used assemblers for bacterial genomes that provide an assembly graph, thus leading to two data sets per isolateâ€¦ xÃ¡c Ä‘á»‹nh xem cÃ¡c contig káº¿t ná»‘i cÃ³ chÃ­nh xÃ¡c lÃ  plasmid khÃ´ng?
3. MÃ¬nh sáº½ láº¥y thÃ´ng tin nÃ y tháº¿ nÃ o (láº¥y trung bÃ¬nh hay sao)

**shaw-2021_cfre-SAMN15148288-s.gfa.csv**

<aside> ğŸ’¡ Platon dá»± Ä‘oÃ¡n Ä‘Æ°á»£c 15/16 plasmid. Pháº§n lá»›n cÃ¡c contig káº¿t ná»‘i vá»›i n thÃ¬ Ä‘Ãºng lÃ  plasmid XÃ¡c Ä‘inh Ä‘Æ°á»£c contig nÃ y lÃ  plasmid vá»›i Ä‘á»™ tá»± tin bao nhiÃªu thÃ¬ káº¿t quáº£ kÃ©o theo sáº½ nhÆ° váº­y

</aside>

# Introduction

There are two major challenges for plasmid identification

1. Plasmids exhibit high genetic diversity
2. The shared genes or segments between plasmids and chromosomes

Tools

1. Graph-based tools
    
    Attempted to reconstruct plasmid underlying assembly graph
    
[https://github.com/cchauve/plASgraph2](https://github.com/cchauve/plASgraph2)
    
Rely on read coverage and cyclic topology for plasmid assembly, which is best used to find relatively comple plasmid rather than short contig
    
2. Alignment-based tools
    
    based on the similarities between queries and reference
    
    <aside> ğŸ’¡ For database we can use vector database to query faster. But we need a way to tokenize the query sequence
    
    </aside>
    
3. Learning-based tools
    
    provide a promising alternative for detecting more diverged plasmids via learning abstract patterns beyond sequence similarity.
    
    Use k-mer freq as the features and applied sequential minimal optimization, fully connected neural network
    
    PPR-Meta encodes the sequences and the contained codons into one-hot matrices and then trains a CNN model for prediction
    
    <aside> ğŸ’¡ Decreased precision on short contigs, whose length limits the capacity of feature learning
    
    </aside>
    
    Can have ambiguous predictions for contigs from shared regions between plasmids and chromosomes
    
4. Hybrid
    
    In order to achieve an optimal trad-off between sensitivity and precision, combine homology search and machine learning to identify as many as possible
    
    PlasmidVerify train a Naive Bayes classifier using protein domain alighment-based feature
    
    PlasForest using homology search results as features
    
    <aside> ğŸ’¡ Also exhibit decreased accuracy on short contigs
    
    </aside>
    
    PLASMe: treat plasmid as a language defined on a vocabulary of protein. Transformer, to learn the protein importantce and their associations for plasmid
    
    > Train for each order
    

# Data

# Alignment

### Embedding

[https://academic.oup.com/mbe/article/32/12/3079/2579220?login=false](https://academic.oup.com/mbe/article/32/12/3079/2579220?login=false)

### Features

- K-mer
- DÃ¹ng log length Ä‘á»ƒ lÃ m gÃ¬ mÃ¬nh khÃ´ng nhá»›
- Relative features
- Additional information encoded in the longer sequences

# Modeling

Káº¿t há»£p theo 3 hÆ°á»›ng

- Alignment
- Graph
- Transforme

**The Graphormer is a graph classification model.**

Transformer for Graphs: An Overview from Architecture Perspective

A Generalization of Transformer Networks to Graphs

Dá»±a trÃªn thÃ´ng tin vá» nhá»¯ng contig xung quanh

ThÃ´ng tin tá»« contig xung quanh gÃ¬ cÃ³ thá»ƒ áº£nh hÆ°á»Ÿng Ä‘áº¿n káº¿t quáº£

# Result

**Performancs of different token sets**

PC-based token achieved the best precision, recall and F1-score

- Using PCs as the tokens allows Transformer to learn the importantce of protein directly
- Can capture the correlation between different proterin on the contig

**Interpreting the performance of Transformer**

- plasmidsâ€™ core genes play an essential role in Transformerâ€™s prediction
- Among of them, 23 have protein function annotations, and 12 are related to plasmid core genes: 5 are about conjugation, 4 about transposases, 2 about partitioning and 1 about replication
- Conjugation proteins are responsible for the horizontal gene transfer between bacteria, while the spread of AMR and virulence factors can help hosts improve their adaptability to the environment.
- The function analysis indicated that these tokens were also associated with the core functions of plasmids.
- However, PC-level tokenization also has some limitations, as it is more prone to out-of-vocabulary (OOV) issues. Therefore, for newer plasmids, using PC tokens may lead to false negatives.

## Future Work

- First, because plasmids have a large diversity, novel plasmids may contain proteins that cannot be aligned with the current database, leading to inaccurate predictions
- Second, the current tokens contain only proteins of plasmids. Studies have shown that proteins critical for bacterial survival are more likely to be found in chromosomes
- Third, the interpretation of the Transformer identified potentially unannotated PCs that may also play an important role in the life of plasmids

<aside> ğŸ’¡ Thay vÃ¬ chá»‰ táº­p trung vÃ o Ä‘áº·c Ä‘iá»ƒm plasmid cÃ³ (gene, protein tren do), tap trung them vao nhung gen co trong bacterial ma khong co trong plasmid (nhÆ° lÃ  hÃ¬nh cá»§a biá»ƒu Ä‘á»“ ven)

</aside>

# Resources

[https://research.google/blog/exphormer-scaling-transformers-for-graph-structured-data/](https://research.google/blog/exphormer-scaling-transformers-for-graph-structured-data/)

- [A Generalization of Transformer Networks to Graphs](https://arxiv.org/pdf/2012.09699)
- [Self-Supervised Graph Transformer on Large-Scale Molecular Data](https://proceedings.neurips.cc/paper_files/paper/2020/file/94aef38441efa3380a3bed3faf1f9d5d-Paper.pdf)
- [TORMES: an automated pipeline for whole bacterial genome analysis](https://academic.oup.com/bioinformatics/article/35/21/4207/5430930)
- [PLASMe: a tool to identify PLASMid contigs from short-read assemblies using transformer](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10450166/pdf/gkad578.pdf)
- [MetaGraph: Plasmid/Chromosome Classification Enhancement Using Graph Neural Networks](https://ieeexplore.ieee.org/document/9906285)
- [plASgraph2: using graph neural networks to detect plasmid contigs from an assembly graph](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2023.1267695/full)
- [3CAC: improving the classification of phages and plasmids in metagenomic assemblies using assembly graphs](https://academic.oup.com/bioinformatics/article/38/Supplement_2/ii56/6702013)
- [Plassembler: an automated bacterial plasmid assembly tool](https://academic.oup.com/bioinformatics/article/39/7/btad409/7208863)
- [Why There Are No Essential Genes on Plasmids](https://academic.oup.com/mbe/article/32/12/3079/2579220?login=false)

[https://github.com/rrwick/Trycycler/wiki/Guide-to-bacterial-genome-assembly#2-short-read-qc](https://github.com/rrwick/Trycycler/wiki/Guide-to-bacterial-genome-assembly#2-short-read-qc)

- Temp
    
    **GNNs as Auxiliary Modules**
    
    - Directly inject GNNs into Transformer architecture.
    - Types
        - Build Transformer blocks on top of GNN blocks
        - Stack GNN blocks and Transformer blocks in each layer
        - Parallel GNNs blocks and Transformer block in each layer
    
    **Improved Positional Embedding from Graphs**
    
    ÄÃ¢y lÃ  dá»± Ä‘á»‹nh ban Ä‘áº§u khi thÃªm features trÆ°á»›c khi sá»­ dá»¥ng Transformer
    
    - Compress the graph structure into positional embedding vectors
    - Add them to the input before it fed to the vanilla Transformer model
    - This graph positional embedding can be derived from the structural information of graphs, such as degree and centrality
    
    **Improved Attention Matrices from Graphs**
    
    **GNNs as Auxiliary Modules**
    
    - Directly inject GNNs into Transformer architecture.
    - Types
        - Build Transformer blocks on top of GNN blocks
        - Stack GNN blocks and Transformer blocks in each layer
        - Parallel GNNs blocks and Transformer block in each layer
    
    **Improved Positional Embedding from Graphs**
    
    ÄÃ¢y lÃ  dá»± Ä‘á»‹nh ban Ä‘áº§u khi thÃªm features trÆ°á»›c khi sá»­ dá»¥ng Transformer
    
    - Compress the graph structure into positional embedding vectors
    - Add them to the input before it fed to the vanilla Transformer model
    - This graph positional embedding can be derived from the structural information of graphs, such as degree and centrality
    
    **Improved Attention Matrices from Graphs**
    
    - Inject graph priors into the attention computation via graph bias terms, or restrict a node only attending to local neighbours in the graph, which can be computationally formulated as an attention masking machanism

[Transformer as a Graph Neural Network â€” DGL 2.2 documentation](https://docs.dgl.ai/en/latest/tutorials/models/4_old_wines/7_transformer.html)

[Transformers are Graph Neural Networks | NTU Graph Deep Learning Lab](https://graphdeeplearning.github.io/post/transformers-are-gnns)

[The Annotated Transformer](https://nlp.seas.harvard.edu/2018/04/03/attention.html)

---
# [Contig](https://www.genome.gov/genetics-glossary/Contig)

A contig (as related to genomic studies; derived from the word â€œcontiguousâ€) is a set of DNA segments or sequences that overlap in a way that provides a contiguous representation of a genomic region. For example, a clone contig provides a physical map of a set of cloned segments of DNA across a genomic region, while a sequence contig provides the actual DNA sequence of a genomic region.

![contig](https://www.genome.gov/sites/default/files/media/images/tg/Contig.jpg)

**â€œReadsâ€ are small segments of nucleic acid sequence produced by a sequencing instrument (there are many kinds).**Â **â€œContigsâ€ represent contiguous sequence formed by assembling the reads**

A genome is typically studied by breaking down its DNA into pieces, analyzing those pieces in the laboratory, and then computationally putting those pieces back together in a conceptual way. Years ago, before genome sequencing became routine, the pieces of DNA would first be cloned and then individual clones would be analyzed to determine which clones overlapped which other clones. A set of overlapping clones that together contain a â€˜contiguousâ€™ region of a genome is called a contig â€“ or in this case a clone contig. Genome sequencing uses the concept of a contig, but in a different way. For genome sequencing, the pieces are not actually physical segments of DNA but rather are stretches of generated DNA sequence (often called a sequence read). By establishing the overlaps among a set of sequence reads, one can assemble a sequence contig, from which the actual sequence of the genomic region can be deduced.

https://www.quora.com/What-are-reads-and-contigs-file-formats-bioinformatics#:~:text=%E2%80%9CReads%E2%80%9D%20are%20small%20segments%20of,formed%20by%20assembling%20the%20reads.
https://www.biostars.org/p/9510018/
https://www.researchgate.net/post/How-to-find-plasmid-sequence-from-contigs

---

### Other tools can help identify plasmid contigs

Kraken

Can assign taxonomy IDs to contigs, indicating whether they are on mobile genetic elements that can be found on chromosomes or plasmid

Plasmer

Can learn from contigs of various lengths by sliding chromosome and plasmid sequences into different window size
https://www.biostars.org/p/9510018/

---

### Structured approach to efficiently retrieve plasmid sequences from contigs, ensuring a comprehensive analysis of sequencing data

The process of identifying plasmid sequences within contigs generated from sequencing projects ==involves== a combination of bioinformatics tools and databases. This approach facilitates the differentiation of chromosomal and plasmid DNA, allowing for the accurate assembly and annotation of plasmid sequences.

Step-by-Step Guide

1. **Quality Assessment and Assembly**:Initially, assess the quality of your raw sequencing data using tools such as FastQC. Utilize assembly software (e.g., SPAdes, Velvet) to generate contigs from your sequencing reads. This process will compile both chromosomal and plasmid-derived sequences.
2. **Contig Classification**:Employ contig classification tools (e.g., PlasmidFinder, PlasFlow) to distinguish between chromosomal and plasmid contigs. These tools utilize databases of known plasmid sequences and machine learning algorithms to predict the origin of contigs.
3. **Database Comparison**:Compare your identified plasmid contigs against plasmid databases (e.g., NCBI's Plasmid Database, PATRIC) using BLAST or similar sequence alignment tools. This step helps to identify known plasmid sequences and assign potential functions to the contigs.
4. **Plasmid Assembly (if necessary)**:For fragmented plasmid sequences scattered across multiple contigs, consider using specialized assembly tools designed for plasmid reconstruction (e.g., plasmidSPAdes) to generate complete plasmid sequences.
5. **Annotation**:Annotate the assembled plasmid sequences using tools such as Prokka or RAST to identify genes, resistance markers, and other functional elements within the plasmid DNA.
6. **Verification and Validation**:Where possible, verify the assembled plasmid sequences through comparison with known plasmids or experimental validation methods, such as PCR, to confirm the presence of specific genes or sequences.

Considerations and Best Practices

- **Database Utilization**: Regularly update your database references to include the latest known plasmid sequences for improved accuracy in classification and annotation.
- **Cross-Validation**: Use multiple tools and databases for contig classification and annotation to cross-validate your results and increase confidence in the identified plasmid sequences.
- **Experimental Confirmation**: Whenever feasible, complement bioinformatic predictions with experimental validation to confirm the existence and structure of plasmids within your samples.