---
tags:
  - Paper
  - Research
links:
  - "[[Plasmid]]"
aliases:
  - Encoding Unitig-level Assembly Graphs with Heterophilous Constraints for Metagenomic Contigs Binning
URL: https://openreview.net/pdf?id=vBw8JGBJWj
---
# Abstract

- Mainstream binning methods primarily rely on sequence features such as composition and abundance, making them unable to effectively handle sequences shorter than 1,000 bp and inherent noise within sequence.
- Using assembly graph generated by assemblers, which encodes valuable overlapping information among genomic sequences
- The contig-level assembly graphs reduces the resolution of the connectivity information in original graphs
- Leverages representation learning on untig-level assembly graphs while adhering to hetorophilous constrains imposed by single-copy marker gene
	- ==Perform opposite== 

# Introduction

- Many metagenomis techniques use assembly graph to assemble these short reads into longer DNA sequences know as unitig
- Many existing metagenomi ... rely on statistical information extracted from contig inclusidn nucleotide composition and abundace features
- These tools do not take into account the homophily information within the assembly graph which suggest that sequences connected to each other in assembly graph are likely to belong to the same species.
- However, the simplification reduces the resolution of the connectivity information in unitig-level assembly graphs and may introduce erroneous edges
- Additional biological information, such as single-copy marker genes, can also be exploited to improve binning results.
- If two contigs share the same single-copy marker gene, it is highly likely that they are belong to differrent species
- There are limited graph neural network model that can be directly employed to model the untig-level assembly graph with heterophil relationships
- Moreover, many existing binning tools face difficulties and exhibit low recall values when ==handling short sequences, often shorter than 1,000== bp, which are commonly excluded from analysis.
- ==single-copy marker genes are genes that occur only once in each species. If two contigs share the same single-copy marker gene, it is highly likely that they belong to different species==

# Related Work

- The majority of existing binners overlook the homophily information present in these assembly graphs. Instead, these binning tools rely on composition and coverage information

## Assembly graph improves binning

- MetaCoAG is a standalong binning tool capable of integating composition, abundace, and assembly graph information to enhance binning performance
- Does not incorporate additional information like hetorophilous constraints from single copy marker genes
- encoding prior constraints
- Use unitig-level assembly

# Methodology

A unitig-level assembly graph G = (V, E, P, X) along with its constraints C
The output embedding for unitigs and contigs in the graph are d-dimensional vector
P corresponds to the path or the contigs within the graph
X are features associated with nodes
Hetoriphilous constrains from the single copy marker genes are C

### Diffusion encoder-decoder framework

An encoder-decoder architecture, which comprises two primary components

- A graph diffusive convolution encoder
	- captures the graph's topology and initial node features
- An inner-product decoder
	- reconstructs the graph's structure using learned features from diffusive encoder

Minimizing the reconstruction loss, which measures the dissimilarity between original and recontructed graph, allow us to obtain the node embeddings.

==Encoder-Decoder==

A variational autoencoder is established

==Diffusive convolution==

Use PageRank-based diffusion to model the unitig-level assembly graph.

### Triplet Gaussian Constraints Optimization

In constraints C, each set signifies that certain contigs contain the identifical marker gene and these contigs must not be grouped pairwise into the same bin.

In Learning, we convert contig-constraints C into the pairwise contig-constraints set C' and pairwise unitig-constrains set M'.

In each pairwise constraint (i, j) from M', it indicates that unitig i and j must not be assigned to the same bin

### Training and Data Batching

Training GNNs on unitig-level assembly graph from real metagenomic data, which can reach milions in size, presents significant computational challenges

Creating training batches present a challenge as it must satisfy two criteria

- Processing each contig in parallel while preserving its completeness
- Grouping diverse contigs with positive relationships into the same batch to retain this valuable information

Graph spliting and training module named P-Batch, which systematically selects independent set of nodes from the Positive contig Graph which derived from the positive contigs set O in an iterative manner.

Takes each path as the minimum splitting unit and functions iteratively through two steps
- Selecting the largeest contigs sets from the candidates
- Feeding them into the smallest batch
The P-Batch will continute until all candidate contigs are fed into one batch. In practice, there are some unitigs that are feed into diffrent batches.. A loss function to minimize the probability distribution of these jointly nodes

## Preprocessing

The heterophilous constrains C are created by employing the FragGeneScan and HMMER tools to detect contigs containing single-copy marker genes, following a similar appraoch as MaxBin ...
- When contigs belong to the same constraint set, it implies that these contigs should not be group together in pairs within the bins
The unitig-level assembly-graph are constructed from two widely used assembler: metaSPAdes and metaFlye

Graph disentangling

GD is designed to separate contigs within the untig-level assembly  graph
When constraints suggest that contig A and contig B should belong to distinct bins, yet these two contigs, share an overlapping unitig in the assembly graph, we create a new unitig by ddiplicating the original unitg 5 to disentange the assmebly graph

# Experiments

# Conclusion